### PR TITLE
Doc deprecated functions for removal in 3.1

### PR DESCRIPTION
**Pull request by jdufresne**
- `ldap.open()`
- `ldif.CreateLDIF()`
- `ldif.ParseLDIF()`

### CODE REVIEW

**Comment by encukou** [comment]
Looks good, thank you!

I'd like to have all deprecation notices also include what to use instead. For example:
> ``ldap.open()`` is deprecated and will be removed in version 3.1. Use `ldap.initialize()` with a URL like `'ldap://<hostname>:<port>'` instead.

While here, I'd also document the `init()` alias (only to say it's deprecated).

I'll can do the changes on Monday before release, if you don't get to them first.

**Comment by jdufresne** [comment]
Sure thing. I updated the PR with suggested alternatives. Please feel free to tweak the wording if you think it could be improved.

File: Lib/ldif.py [hunk 1]
```diff
@@ -209,7 +209,7 @@ def unparse(self,dn,record):
 def CreateLDIF(dn,record,base64_attrs=None,cols=76):
   """
   Create LDIF single formatted record including trailing empty line.
-  This is a compatibility function. Use is deprecated!
+  This is a compatibility function.
```
Comment (tiran) [inline Lib/ldif.py]: ``warnings.warn('CreateLDIF is deprecated, use LDIFWriter.unparse() instead.', category=DeprecationWarning, stacklevel=2)``

File: Lib/ldif.py [hunk 2]
```diff
@@ -633,7 +633,7 @@ def handle(self,dn,entry):
 def ParseLDIF(f,ignore_attrs=None,maxentries=0):
   """
   Parse LDIF records read from file.
-  This is a compatibility function. Use is deprecated!
+  This is a compatibility function.
```
Comment (tiran) [inline Lib/ldif.py]: ``warnings.warn('ParseLDIF is deprecated, use LDIFRecordList.parse() instead.', category=DeprecationWarning, stacklevel=2)``

**Comment by jdufresne** [comment]
Added warnings. I also added "It will be removed in python-ldap 3.1." to all warnings.

**Comment by encukou** [comment]
Thank you!

**Pull request merged**